sqlglot/parser.py

```diff
@@ -1804,27 +1806,39 @@ def _parse_types(self):
             if not self._match(TokenType.GT):
                 self.raise_error("Expecting >")
 
+        value = None
         if type_token in self.TIMESTAMPS:
-            tz = self._match(TokenType.WITH_TIME_ZONE) or type_token == TokenType.TIMESTAMPTZ
-            if tz:
-                return exp.DataType(
+            if self._match(TokenType.WITH_TIME_ZONE) or type_token == TokenType.TIMESTAMPTZ:
+                value = exp.DataType(
                     this=exp.DataType.Type.TIMESTAMPTZ,
                     expressions=expressions,
                 )
-            ltz = (
+            elif (
                 self._match(TokenType.WITH_LOCAL_TIME_ZONE) or type_token == TokenType.TIMESTAMPLTZ
-            )
-            if ltz:
-                return exp.DataType(
+            ):
+                value = exp.DataType(
                     this=exp.DataType.Type.TIMESTAMPLTZ,
                     expressions=expressions,
                 )
-            self._match(TokenType.WITHOUT_TIME_ZONE)
+            else:
+                self._match(TokenType.WITHOUT_TIME_ZONE)
+                value = exp.DataType(
+                    this=exp.DataType.Type.TIMESTAMP,
+                    expressions=expressions,
+                )
 
-            return exp.DataType(
-                this=exp.DataType.Type.TIMESTAMP,
-                expressions=expressions,
-            )
+        if maybe_func and check_func:
+            index2 = self._index
+            peek = self._parse_column()
```

Comment:
any way we can do an actual peek here instead of a full column parse?  

can you just check the token type of self._curr token? what is the case where we don’t have a column

what is the case where you know

Reply:
after looking, every case I've found of a type literal that optionally takes a modifier - so it's ambiguous whether it's a function - has a string for the expression